plugins/FirmwareUpdateChecker/tests/TestFirmwareUpdateChecker.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+
+import pytest
+import json
+
+import unittest.mock
+from cura.CuraApplication import CuraApplication
+
+from UM.Version import Version
+
+from plugins.FirmwareUpdateChecker.FirmwareUpdateCheckerJob import FirmwareUpdateCheckerJob
+from plugins.FirmwareUpdateChecker.FirmwareUpdateCheckerLookup import FirmwareUpdateCheckerLookup
+
+json_data = \
+    {
+        "ned":
+            {
+                "id": 1,
+                "name": "ned",
+                "check_urls": ["http://urlecho.appspot.com/echo?status=200&body=5.1.2.3"],
+                "update_url": "https://ultimaker.com/en/resources/20500-upgrade-firmware",
+                "version_parser": "default"
+            },
+        "olivia":
+            {
+                "id": 3,
+                "name": "olivia",
+                "check_urls": ["http://urlecho.appspot.com/echo?status=200&body=4.3.2.1", "https://this-url-clearly-doesnt-exist.net"],
+                "update_url": "https://ultimaker.com/en/resources/20500-upgrade-firmware",
+                "version_parser": "default"
+            },
+        "emmerson":
+            {
+                "id": 5,
+                "name": "emmerson",
+                "check_urls": ["http://urlecho.appspot.com/echo?status=200&body=0.2.2.2", "http://urlecho.appspot.com/echo?status=200&body=6.7.8.1"],
```

Comment:
I'm not quite sure about this (although I don't quite have a better way of doing it), but this will fail the unit test if there is no internet connection.

Reply:
You're right. I could have a look to see if it works with 'file://', since it seems that is supported (even for 2, despite the misleading name): https://stackoverflow.com/questions/20558587/opening-local-file-works-with-urllib-but-not-with-urllib2

Reply:
Update: No dice; 'file://...' (like the other schemes) requires absolute paths, so it would have to be different for every builder.